File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0;
        while(i < nums.length) {
            int j = i + 1;
            while(j < nums.length && nums[j] == nums[i]) {
                j++;
            }

            if(j == nums.length || nums[i] >= nums[j]) {
                return i+1;
            }

            //shift elements
            int p = i +1;
            while(j < nums.length) {
                nums[p] = nums[j];
                p++;
                j++;
            }

            i++;
        }

        return i+1;
    }
}

/**

1 2 2 2 2 2 2 2

**/

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [0] * len(nums)
        suffix = [0] * len(nums)

        curr = 1
        for i in range(len(nums)):
            prefix[i] = curr 
            curr *= nums[i]

        curr = 1
        for i in range(len(nums) - 1,-1,-1):
            suffix[i] = curr
            curr *= nums[i]

        output = [0] * len(nums)
        for i in range(len(nums)):
            output[i] = prefix[i] * suffix[i]

        return output    


            